src/main/java/mate/academy/rickandmorty/service/RickAndMortyClient.java

```diff
@@ -0,0 +1,37 @@
+package mate.academy.rickandmorty.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.rickandmorty.dto.CharacterResponseDto;
+import mate.academy.rickandmorty.dto.ListCharacterDto;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class RickAndMortyClient {
+    private static final String CHARACTER_URL = "https://rickandmortyapi.com/api/character";
+    private final ObjectMapper objectMapper;
+
+    public List<CharacterResponseDto> getCharacters() {
+        HttpClient httpClient = HttpClient.newHttpClient();
```

Comment:
Make HttpClient the spring bean. 
It shouldnt be created every method invokation.
```suggestion
        HttpClient httpClient = HttpClient.newHttpClient();
```